CRM/Core/BAO/FinancialTrxn.php

```diff
@@ -540,4 +540,21 @@ public static function getTotalPayments($contributionId) {
     return CRM_Core_DAO::singleValueQuery($sql, $params);
   }
 
+  /**
+   * @param array $contribution
+   * @param array $params
+   *
+   * @return CRM_Core_BAO_FinancialTrxn
+   */
+  public static function getPartialPaymentTrxn($contribution, $params) {
```

Comment:
Is the name of this function right? The function name is 'getPartialPaymentTrxn' - but actually it's taking a whole lot of actions. That seems pretty confusing to me.

As an aside - the full spec for code standards includes some text in the comment block explaining what the function does. The first line is supposed to be a sentence that fits on one line & starts with a capital & ends with a full stop.